### PR TITLE
feat(docs): add markdown based documentation

### PR DESCRIPTION
**Pull request by swarit-pandey**

### CODE REVIEW

**Comment by swarit-pandey** [comment]
We are yet to decide on which license to use for this github action. 
cc: @nyrahul

File: README.md [hunk 1]
```diff
@@ -1,3 +1,190 @@
...
+
+## KubeArmor
+KubeArmor is a CNCF project that powers this action to scan the events taking place
+in the CI/CD environment such as GitHub runner. KubeArmor uses eBPF monitors to 
+watch over the events taking place at system level and uses LSMs (Linux security modules)
+for enforcing the security policies either in block or audit mode. 
+
+To learn more about KubeArmor please visit, https://kubearmor.io/ 
+
+We install KubeArmor in systemd mode in the GitHub runner and that lets us watch over
```
Comment (nyrahul) [inline README.md]:
```suggestion
This Action installs KubeArmor in systemd mode in the GitHub runner and watches over
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,190 @@
-# report-action
+# CI/CD Security by AccuKnox
```
Comment (nyrahul) [inline README.md]: # Monitor & Audit CI/CD pipelines

File: README.md [hunk 3]
```diff
@@ -1,3 +1,190 @@
...
+- [Contributing](#contributing)
+
+## Introduction
+The CI/CD Scan by AccuKnox is a powerful tool designed to enhance the security of 
+your continuous integration and deployment pipelines. By providing comprehensive 
+assessment, monitoring, and protection capabilities, it enables you to:
+
+- Detect and prevent potential supply chain attacks in CI/CD environments
+- Gain real-time visibility into your pipeline's security posture
+- Streamline the integration of security practices into your DevOps workflow
```
Comment (nyrahul) [inline README.md:22]:
```suggestion
- Detect and prevent potential supply chain attacks in CI/CD environments
- Gain real-time visibility into your pipeline's security posture
- Streamline the integration of security practices into your DevOps workflow

Ensure application best practices by:
* Applying app hardening policies and checking whether it deviates during GH workflow execution.
* Identifying if there are any unknown processes spawning during CI/CD workflow execution.
* Identifying if any unwanted network connections are started in the pipeline.
```

File: README.md [hunk 4]
```diff
@@ -1,3 +1,190 @@
...
+
+It clearly tells you about the policy itself, the command which caused the generation of alert
+and the message itself, which makes it easy for you to understand the nature of alert and
+a possible issue in the CI/CD pipeline itself.
+
+### 2) Define your own security policies
+Other than providing the baseline as an initial security posture, you can also write your
+own security policies and have a more granular control on the CI/CD pipeline. 
+
+Here is an example of a security policy
```
Comment (nyrahul) [inline README.md:71]: Remove the below YAML policy and provide a realistic policy from existing template

File: README.md [hunk 5]
```diff
@@ -1,3 +1,190 @@
...
+
+### 4) Network events
+You can also see all the network based events taking place in the system, this 
+is very important to identify any egress or ingress network events with the network
+protocol, process making the network call and the name of the process itself. 
+
+For example:
+![](./network_example_1.png)
+
+<em>We will also support standard network policies in the coming realeases.</em>
```
Comment (nyrahul) [inline README.md]:
```suggestion
<em>Support for network policies in the coming releases.</em>
```

File: README.md [hunk 6]
```diff
@@ -1,3 +1,190 @@
...
+
+## Usage
+The usage is as simple as <em>Plug-and-Play</em>, you only have to include the following 
+lines in your GitHub workflow and you are all set to go. 
+
+```yaml
+- name: AccuKnox CI/CD scan
```
Comment (nyrahul) [inline README.md]:
```suggestion
- name: AccuKnox CI/CD Monitor
```

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (swarit-pandey) [reply README.md]: Addressed.

File: README.md (hunk 2 above)
Comment (swarit-pandey) [reply README.md]: Addressed.

File: README.md (hunk 3 above)
Comment (swarit-pandey) [reply README.md:22]: Addressed.

File: README.md (hunk 4 above)
Comment (swarit-pandey) [reply README.md:71]: Added a more realistic policy.

File: README.md (hunk 5 above)
Comment (swarit-pandey) [reply README.md]: Addressed.

File: README.md (hunk 6 above)
Comment (swarit-pandey) [reply README.md]: Addressed.

**Pull request merged**